src/main/java/io/cloudtrust/keycloak/eventemitter/EventEmitterProvider.java

```diff
@@ -125,27 +151,37 @@ private void sendEventsWithJsonFormat() {
         for (int i=0; i < pendingEventsSize; i++){
             IdentifiedEvent event = pendingEvents.poll();
 
+            if(event == null){
```

Comment:
You can avoid to call pendingEvents.size() and avoid using break :
```
while ((event=pendingEvents.poll())!=null) {
   ...
}
```
Or :
```
for(IdentifiedEvent event=pendingEvents.poll(); event!=null; event=pendingEvents.poll()) {
    ...
}
```

Reply:
If we have a continuous flow of new thread which creates new events added to the queue. We might have a thread that will loop and consume all of them.

Reply:
ok